src/base.jl

```diff
@@ -17,24 +18,54 @@ function check_size(T, sz)
     end
 end
 
-function ShortString{T}(s::Union{String, SubString{String}}) where {T}
-    sz = sizeof(s)
-    check_size(T, sz)
-    bits_to_wipe = 8(sizeof(T) - sz)
+size_bytes(::Type{T}) where {T} = (count_ones(sizeof(T)-1)+7)>>3
 
-    # Warning: if a SubString is at the very end of a string, which is at the end of allocated
-    # memory, this can cause an access violation, by trying to access past the end
-    # (for example, reading a 1 byte substring at the end of a length 119 string, could go past
-    # the end)
+size_mask(T) = T((1<<(size_bytes(T)*8)) - 1)
+size_mask(s::ShortString{T}) where {T} = size_mask(T)
 
-    # TODO some times this can throw errors for longish strings
-    # Exception: EXCEPTION_ACCESS_VIOLATION at 0x1e0b7afd -- bswap at C:\Users\RTX2080\.julia\packages\BitIntegers\xU40U\src\BitIntegers.jl:332 [inlined]
-    # ntoh at .\io.jl:541 [inlined]
-    content = (T(s |> pointer |> Ptr{T} |> unsafe_load |> ntoh) >> bits_to_wipe) << bits_to_wipe
-    ShortString{T}(content | T(sz))
+const CHUNKSZ   = sizeof(UInt)
+const CHUNKBITS = sizeof(UInt) == 4 ? 32 : 64
+
+@inline function _ss(::Type{T}, str::String, sz) where {T}
```

Comment:
Why not write:
```suggestion
@inline function _ss(T::Type, str::String, sz)
```

Also would be good with a docstring or a comment explaining what it is.

Reply:
I originally just had `T`, but then worried that Julia might not specialize on that, didn't have the time to double check.
I'll had the docstring in the next PR!